nix/modules/nixos/holo-host-agent.nix

```
@@ -64,6 +70,41 @@ in
         default = "${cfg.nats.listenHost}:${builtins.toString cfg.nats.listenPort}";
       };
 
+      nscPath = lib.mkOption {
+        type = lib.types.path;
+        default = "/var/lib/.local/share/nats/nsc";
+      };
+
+      sharedCredsPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.nscPath}/shared_creds";
+      };
+
+      localCredsPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.nscPath}/local_creds";
+      };
+
+      hostNkeyPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.localCredsPath}/host.nk";
+      };
+
+      sysNkeyPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.localCredsPath}/sys.nk";
+      };
+
+      hposCredsPath = lib.mkOption {
+        type = lib.types.path;
+        default = "/var/lib/server-key-config.json";
+      };
+
+      hposCredsPw = lib.mkOption {
+        type = lib.types.str;
+        default = "pass";
+      };
```

```suggestion
      hposCredsPwFile = lib.mkOption {
        type = lib.types.path;
        default = "/var/lib/holo-host-agent/hpos_creds_pw";
      };
```

the path could be somewhere else. my intention here is to not hardcode some default password. for placing this file accordingly we'll need a different mechanism.